docs/azure/sdk/dependency-injection.md

```diff
@@ -106,43 +116,97 @@ In the [Register client](#register-client) section, you explicitly specify the `
+
+* Represents the retry policy configuration settings.
+* Corresponds to the <xref:Azure.Core.ClientOptions.Retry> property. Within that object literal, you find the `MaxRetries` key, which corresponds to the <xref:Azure.Core.RetryOptions.MaxRetries> property.
 
-Since the `Configuration` object is a member of the `builder` instance, you can configure secrets:
+For more information, see [Configure a new retry policy](#configure-a-new-retry-policy).
+
+The settings in the JSON configuration file can be retrieved using <xref:Microsoft.Extensions.Configuration.IConfiguration>.
+
+### [WebApplicationBuilder](#tab/web-app-builder)
 
 ```csharp
 builder.Services.AddAzureClients(clientBuilder =>
 {
-    // Add a KeyVault client
-    clientBuilder.AddSecretClient(builder.Configuration.GetSection("KeyVault"));
+    clientBuilder.AddSecretClient(
+        builder.Configuration.GetSection("KeyVault"));
 
-    // Add a storage account client
-    clientBuilder.AddBlobServiceClient(builder.Configuration.GetSection("Storage"));
+    clientBuilder.AddBlobServiceClient(
+        builder.Configuration.GetSection("Storage"));
 
-    // Use DefaultAzureCredential by default
     clientBuilder.UseCredential(new DefaultAzureCredential());
 
     // Set up any default settings
-    clientBuilder.ConfigureDefaults(builder.Configuration.GetSection("AzureDefaults"));
+    clientBuilder.ConfigureDefaults(
+        builder.Configuration.GetSection("AzureDefaults"));
 });
+```
 
-builder.Services.AddControllers();
+### [HostApplicationBuilder](#tab/host-app-builder)
+
+```csharp
+builder.Services.AddAzureClients(clientBuilder =>
+{
+    clientBuilder.AddSecretClient(
+        builder.Configuration.GetSection("KeyVault"));
+
+    clientBuilder.AddBlobServiceClient(
+        builder.Configuration.GetSection("Storage"));
+
+    clientBuilder.UseCredential(new DefaultAzureCredential());
+
+    // Set up any default settings
+    clientBuilder.ConfigureDefaults(
+        builder.Configuration.GetSection("AzureDefaults"));
+});
 ```
 
+### [HostBuilder](#tab/host-builder)
+
+```csharp
+IHost host = Host.CreateDefaultBuilder(args)
+    .ConfigureServices((hostContext, services) =>
+    {
+        services.AddHostedService<Worker>();
+        services.AddAzureClients(clientBuilder =>
+        {
+            clientBuilder.AddSecretClient(
+                hostContext.Configuration.GetSection("KeyVault"));
+
+            clientBuilder.AddBlobServiceClient(
+                hostContext.Configuration.GetSection("Storage"));
+
+            clientBuilder.UseCredential(new DefaultAzureCredential());
+
+            // Set up any default settings
+            clientBuilder.ConfigureDefaults(
+                hostContext.Configuration.GetSection("AzureDefaults"));
+        });
+    })
+    .Build();
+```
+
+---
+
 ## Configure multiple service clients with different names
```

Comment:
I think it would be great to add a section for "registering sub-clients", which are starting to become more common, and builds on the "named client" pattern.  The `ServiceBusSender` scenario would work well for illustration here.   We can cut out the parts where we query the available queues and just register two named instances.   The important part is using the provider to query the parent client.

Reply:
I'll address this and https://github.com/dotnet/docs/issues/34621 in a separate PR.